implementations/micrometer-registry-elastic/src/main/java/io/micrometer/elastic/ElasticMeterRegistry.java

```diff
@@ -254,9 +256,23 @@ private String indexName() {
 
     // VisibleForTesting
     Optional<String> writeMeter(Meter meter) {
+        Iterable<Measurement> measurements = meter.measure();
+        List<String> names = new ArrayList<>();
+        List<Double> values = new ArrayList<>();
+        for (Measurement measurement : measurements) {
+            double value = measurement.getValue();
+            if (!Double.isFinite(value)) {
+                continue;
+            }
+            names.add(measurement.getStatistic().getTagValueRepresentation());
+            values.add(value);
+        }
+        if (names.isEmpty()) {
+            return Optional.empty();
+        }
```

Comment:
The current implementation is correct and works, but I wondered if maybe something like the following is more straightforward.

```java
List<Measurement> measurements = StreamSupport.stream(meter.measure().spliterator(), false)
        .filter(measurement -> Double.isFinite(measurement.getValue()))
        .collect(Collectors.toList());
if (measurements.isEmpty()) {
    return Optional.empty();
}
```

Then the code below can stay the same also.

Reply:
@shakuzen The reason I chose that way is due to the possibility of changes on the values in-between as you pointed it out in another issue.

Reply:
Oh how short my memory is :D
We should probably leave a comment to make it more apparent why the implementation is that way, then, so someone like forgetful me doesn't reintroduce a bug like that.

Reply:
@shakuzen Thanks for the feedback! I added a comment in 1f78437867c60ff994a5f430d044310f5e684af1.